x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexService.java

```diff
@@ -121,19 +142,30 @@ protected static class TransportRequest extends AbstractLookupService.TransportR
             Source source
         ) {
             super(sessionId, shardId, inputDataType, inputPage, toRelease, extractFields, source);
+            this.configuration = configuration;
             this.matchField = matchField;
         }
 
         static TransportRequest readFrom(StreamInput in, BlockFactory blockFactory) throws IOException {
             TaskId parentTaskId = TaskId.readFromStream(in);
             String sessionId = in.readString();
+            Configuration configuration = null;
+            if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_LOOKUP_JOIN_CONFIGURATION_IN_REQUEST)) {
+                configuration = new Configuration(
+                    // TODO make EsqlConfiguration Releasable
+                    new BlockStreamInput(
+                        in,
+                        new BlockFactory(new NoopCircuitBreaker(CircuitBreaker.REQUEST), BigArrays.NON_RECYCLING_INSTANCE)
```

Comment:
Configuration can have some pretty heavy stuff in it. I wonder if we should include only the `Source`. That's what this is about, right?

Reply:
Sending the `Source`?

Reply:
The way `Source.readFrom(planIn);` works, is that the serialized source only contains the line/column numbers, and then it gets the text source from the configuration. So my first go-to was adding the configuration to the PlanStreamInput.

But I can surely send the text separately, read the source (Which won't have a text), and then add the text manually.
A bit hacky, but should work well and should be safe. Gonna do that